src/filterBuilder.js

```diff
@@ -0,0 +1,135 @@
+
+
+export class FilterBuilder {
+
+    STATUS_SCHEDULED = 'scheduled';
+
+    constructor(
+                filter = '',
+                {
+                    user = undefined,
+                    availableContentStatuses = [],
+                    bypassPermissions = false,
+                    pullFutureContent = false,
+                    getFutureContentOnly = false,
+                    getFutureScheduledContentsOnly = false,
+
+                }={}) {
+        this.user = user;
+        this.availableContentStatuses = availableContentStatuses;
+        this.bypassPermissions = bypassPermissions;
+        this.pullFutureContent = pullFutureContent;
+        this.getFutureContentOnly = getFutureContentOnly;
+        // TODO should getFollowedContentOnly be re-added. Currently prod this is only based on subscribed coaches.
+        // If so update _applyFollowedContentOnly()
+        // this.getFollowedContentOnly = getFollowedContentOnly;
+        this.getFutureScheduledContentsOnly = getFutureScheduledContentsOnly;
+        this.filter = filter;
+        this.debug = true;
```

Comment:
This would be nice to have read from the .env
```suggestion
        this.debug = process.env.DEBUG === 'true' || false;
```
(note I didn't test this suggested change)